Math/NumberTheory/ArithmeticFunctions/Standard.hs

```diff
@@ -149,6 +151,23 @@ jordanHelper pa 2 = (pa - 1) * pa
 jordanHelper pa k = (pa - 1) * pa ^ wordToInt (k - 1)
 {-# INLINE jordanHelper #-}
 
+ramanujan :: (UniqueFactorisation n, Integral n, Integral m) => n -> m
+ramanujan = runFunction ramanujanA
+
+-- | Calculates the <https://en.wikipedia.org/wiki/Ramanujan_tau_function Ramanujan tau function>
+--   of a positive number @n@, using formulas given <http://www.numbertheory.org/php/tau.html here>
+ramanujanA :: forall n m. (UniqueFactorisation n, Integral n, Integral m) => ArithmeticFunction n m
+ramanujanA = multiplicative $ \((unPrime :: Prime n -> n) -> p) -> ramanujanHelper p
+
+ramanujanHelper :: (UniqueFactorisation n, Integral n, Integral m) => n -> Word -> m
+ramanujanHelper _ 0 = 1
+ramanujanHelper 2 1 = -24
+ramanujanHelper p 1 = (65 * sigmaHelper (p ^ (11 :: Int)) 1 + 691 * sigmaHelper (p ^ (5 :: Int)) 1 - 691 * 252 * 2 * sum [sigma 5 k * sigma 5 (p-k) | k <- [1..(p `quot` 2)]]) `quot` 756
+ramanujanHelper p k = sum [(-1)^j * binomial !! (wordToInt $ k-j) !! (wordToInt $ k - 2*j) * pa^j * tp ^ (k - 2*j) | j <- [0..k `quot` 2]]
```

Comment:
I wonder whether this can be improved to avoid exponentiation and random access to a list. Haskell is not clever enough to optimise even `(-1)^j`, not saying about succeeding powers of `pa` and decreasing powers of `tp`. And indexing of a list has linear complexity.

One can write something like 
```
minusOnePowers = cycle [1, -1]
paPowers = iterate (* pa) 1 
tpPowers = reverse $ iterate (* tp ^ 2) (if even k then 1 else tp)
binomials = map (\j -> binomial !! (wordToInt $ k-j) !! (wordToInt $ k - 2*j)) [0 .. k `quot` 2]
```
and zip them altogether 
```
zipWith4 (\a b c d -> a * b * c * d) minusOnePowers paPowers tpPowers binomials
```

This avoids exponentiation, but not random access. I do not know whether there is a clever trick to avoid the latter.

Reply:
Ah, I (mistakenly) hoped exponentiation would be optimised like that. I'll push with new benchmarks. I couldn't see a nicer way of doing the binomial indexing either - other than that if `j` is in a certain range, computing `k-j` choose `j` is faster.

Reply:
1. `binomial !! (k-j) !! (k-2*j) = binomial !! (k-j) !! j`, where for `j <- [0.. k / 2]` the latter expression looks simpler. 

2. We can always resort to [factorials](https://github.com/cartazio/arithmoi/blob/master/Math/NumberTheory/Recurrencies/Linear.hs#L34). The binomial sequence in question equals to `(k-j)! / (k-2j)! / j!`, which can be built up in a following way:

```haskell
zipWith (\a b c -> a `quot` b `quot` c) xs (skipOdds xs) factorial
  where
    xs = reverse $ take k factorial  
```

`skipOdds` can be copied from [Math.NumberTheory.Zeta](https://github.com/cartazio/arithmoi/blob/master/Math/NumberTheory/Zeta.hs#L41). 

I do not know, whether this factorial approach is faster than indexing